### PR TITLE
Add converter from SonarQubeIssue to AnalysisIssueVisualization

### PR DESCRIPTION
**Pull request by rita-g-sonarsource**
Part of #1988

### CODE REVIEW

File: src/IssueViz.Security/Taint/SonarQubeIssueToIssueVisualizationConverter.cs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                    location.TextRange.EndLine,
+                    location.TextRange.StartOffset,
+                    location.TextRange.EndOffset,
+                    null);
+            }).ToArray();
+
+        /// <summary>
+        /// Converts from the sonarqube issue severity enum to the standard AnalysisIssueSeverity
+        /// </summary>
+        internal /* for testing */ static AnalysisIssueSeverity Convert(SonarQubeIssueSeverity issueSeverity)
```
Comment (rita-g-sonarsource) [inline src/IssueViz.Security/Taint/SonarQubeIssueToIssueVisualizationConverter.cs:107]: Identical to https://github.com/SonarSource/sonarlint-visualstudio/blob/master/src/Integration.Vsix/SonarLintDaemon/SonarLintDaemon.cs#L389

File: src/IssueViz.Security/Taint/SonarQubeIssueToIssueVisualizationConverter.cs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        {
+            var analysisIssue = ConvertToAnalysisIssue(sonarQubeIssue);
+            var issueViz = issueVisualizationConverter.Convert(analysisIssue);
+
+            return issueViz;
+        }
+
+        private IAnalysisIssueBase ConvertToAnalysisIssue(SonarQubeIssue sonarQubeIssue)
+        {
+            if (sonarQubeIssue.TextRange == null)
```
Comment (rita-g-sonarsource) [inline src/IssueViz.Security/Taint/SonarQubeIssueToIssueVisualizationConverter.cs:60]: I don't think it's possible to have a vulnerability without a span or to have a location without a span, but I'll check tomorrow with Alex/Marco.

File: src/IssueViz.Security/Taint/SonarQubeIssueToIssueVisualizationConverter.cs [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            {
+                throw new ArgumentNullException(nameof(sonarQubeIssue.TextRange));
+            } 
+
+            var localFilePath = absoluteFilePathLocator.Locate(sonarQubeIssue.FilePath);
+
+            return new AnalysisIssue(
```
Comment (rita-g-sonarsource) [inline src/IssueViz.Security/Taint/SonarQubeIssueToIssueVisualizationConverter.cs:67]: I'm using the regular AnalysisIssue for now (like we do in the deamon and cfamily code). I didn't see the need for a dedicated Taint class (yet).

File: src/IssueViz.Security/Taint/SonarQubeIssueToIssueVisualizationConverter.cs [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+                sonarQubeIssue.RuleId,
+                Convert(sonarQubeIssue.Severity),
+                AnalysisIssueType.Vulnerability,
```
Comment (duncanp-sonar) [inline src/IssueViz.Security/Taint/SonarQubeIssueToIssueVisualizationConverter.cs:70]: There's nothing in the naming of this interface/class to suggest that it's specific to vulnerabilities. If it's a general purpose class it should handle all issue types.

File: src/IssueViz.Security/Taint/SonarQubeIssueToIssueVisualizationConverter.cs (hunk 1 above)
Comment (duncanp-sonar) [reply src/IssueViz.Security/Taint/SonarQubeIssueToIssueVisualizationConverter.cs:107]: I don't think it's worth introducing a separate converter for this. However, the duplication isn't good either - a shared static class+method would be better.

**Review by duncanp-sonar — COMMENTED** [review]
There are a couple of things I think should be tweaked; the class shouldn't have a general-purpose name if it's hard-coded to assume the issue is a vulnerability, and the code duplication could be avoided.

File: src/IssueViz.Security/Taint/SonarQubeIssueToIssueVisualizationConverter.cs (hunk 1 above)
Comment (rita-g-sonarsource) [reply src/IssueViz.Security/Taint/SonarQubeIssueToIssueVisualizationConverter.cs:107]: I'm sorry I should've clarified. It's identical in terms of logic, it's different enums. SonarQubeIssueSeverity vs DaemonIssueType.

Comment (duncanp-sonar) [reply src/IssueViz.Security/Taint/SonarQubeIssueToIssueVisualizationConverter.cs:107]: And the 'DaemonIssueType' is part of the gRPC message format so we can't share a single enum. Ok.

File: src/IssueViz.Security/Taint/SonarQubeIssueToIssueVisualizationConverter.cs (hunk 4 above)
Comment (rita-g-sonarsource) [reply src/IssueViz.Security/Taint/SonarQubeIssueToIssueVisualizationConverter.cs:70]: You're right. I renamed it to `TaintIssueToIssueVisualizationConverter` for now, as it also verifies that there is a location (so it's definitely not right for all issues).

**Review by duncanp-sonar — APPROVED** [review]

**Pull request merged**